python/cuml/common/base.pyx

```diff
@@ -364,6 +367,14 @@ class Base(TagsMixin,
             return {'preserves_dtype': [self.dtype]}
         return {}
 
+    def _set_nvtx_annotations(self):
```

Comment:
we should measure impact and perhaps add a way to make this a no-op if pertinent

Reply:
I've computed an average of several runs of a cuML estimator with and without setting nvtx annotations on `fit` and `transform`. It appears the difference I measured is only of a tenth of a millisecond. Might still be relevant to add a parameter to the `Base` class to enable/disable it. What do you think?

Reply:
Actually, adding a parameter might not be the best idea as it would require modifying the code of every model. I noticed that the pickling feature stopped working when methods were overwritten after class initialization. Because of this, I believe the best option is to manually call the `set_nvtx_annotations` method whenever we want to enable some benchmarking. This should also prevent any slowdown due to NVTX marking.